src/plugins/intel_gpu/src/kernel_selector/kernels/reorder/reorder_weights_opt.cpp

```diff
@@ -55,6 +55,39 @@ ParamsKey ReorderWeightsOpt::GetSupportedKey() const {
     return k;
 }
 
+DeviceFeaturesKey ReorderWeightsOpt::get_required_device_features_key(const Params& params, const optional_params& /*options*/) const {
+    DeviceFeaturesKey k;
+
+    bool requires_blocked_read_write_char = false;
+    bool requires_blocked_read_write_short = false;
+    bool requires_blocked_read_write = false;
+    const auto& casted_params = static_cast<const reorder_weights_params&>(params);
+
+    std::vector<WeightsType> tensor_types = {casted_params.input.GetDType(), casted_params.output.GetDType() };
+    for (auto& type : tensor_types) {
+        if (type == WeightsType::F16) {
+            requires_blocked_read_write_short = true;
+        } else if (type == WeightsType::F32) {
+            requires_blocked_read_write = true;
+        } else if (type == WeightsType::UINT8 || type == WeightsType::INT8) {
+            requires_blocked_read_write_char = true;
+        }
+    }
+
+    if (requires_blocked_read_write)
+        k.requires_blocked_read_write();
+
+    if (requires_blocked_read_write_short)
+        k.requires_blocked_read_write_short();
+
+    if (requires_blocked_read_write_char)
+        k.requires_blocked_read_write_char();
+
+    k.requires_subgroups();
```

Comment:
Can be replaced with `get_common_subgroups_device_features_key()`

Reply:
`get_common_subgroups_device_features_key()` does `Params` cast to `base_params` internally, `reorder_weights_params` can't be casted to it.